kustomize/commands/edit/fix/convert.go

```diff
@@ -134,7 +135,7 @@ func getNodesFromFile(fileName string, fSys filesys.FileSystem) ([]*kyaml.RNode,
 	}
 	out := &bytes.Buffer{}
 	r := kio.ByteReadWriter{
-		Reader:                bytes.NewBufferString(string(b)),
+		Reader:                bytes.NewReader(b),
```

Comment:
Please add a code comment here that the reason we have to change to `NewReader()`

Reply:
The `NewReader()` can directly create `io.Reader` from `[]byte`, I think it is redundant to convert a `[]byte` to `string` then back to `[]byte`. 

Internally, the `NewBufferString` will just convert `string` back to `[]byte`.

```golang
func NewBuffer(buf []byte) *Buffer { return &Buffer{buf: buf} }

func NewBufferString(s string) *Buffer {
	return &Buffer{buf: []byte(s)}
}
```

Reply:
I got your point; I have the same idea here for the code, but we cannot see more useful from `NewReader(b)`, I want to make sure it will not cause any other issues. So here we need to add the testcase.

Reply:
Ok. So do I need to add test cases directly for `getNodesFromFile`?

I've add a test case for the nil pointer dereferencing including `writePatchTargets`